### PR TITLE
dreambooth if docs - stage II, more info

### PR DESCRIPTION
**Pull request by williamberman**
More fleshed out docs we can link to in model cards and IF readme

### CODE REVIEW

File: docs/source/en/training/dreambooth.mdx [hunk 1]
```diff
@@ -502,9 +502,49 @@ You may also run inference from any of the [saved training checkpoints](#inferen
...
+
+```py
+from diffusers import DiffusionPipeline
+
+pipe = DiffusionPipeline.from_pretrained("DeepFloyd/IF-I-XL-v1.0")
+
+pipe.load_lora_weights("<lora weights path>")
+
+# Update scheduler config to fixed variance schedule
+pipe.scheduler = pipe.scheduler.__class__.from_config(pipe.scheduler.config, variance_type="fixed_small")
```
Comment (sayakpaul) [inline docs/source/en/training/dreambooth.mdx:520]: 🧠

File: docs/source/en/training/dreambooth.mdx [hunk 2]
```diff
@@ -502,9 +502,49 @@ You may also run inference from any of the [saved training checkpoints](#inferen
...
+```
+
+Additionally, a few alternative cli flags are needed for IF.
+
+`--resolution=64`: IF is a pixel space diffusion model. In order to operate on un-compressed pixels, the input images are of a much smaller resolution. 
+
+`--pre_compute_text_embeddings`: IF uses T5 for its text encoder. In order to save GPU memory, we pre compute all text embeddings and then de-allocate
```
Comment (sayakpaul) [inline docs/source/en/training/dreambooth.mdx:527]:
```suggestion
`--pre_compute_text_embeddings`: IF uses [T5](https://huggingface.co/docs/transformers/model_doc/t5) for its text encoder. In order to save GPU memory, we pre compute all text embeddings and then de-allocate
```

File: docs/source/en/training/dreambooth.mdx [hunk 3]
```diff
@@ -502,9 +502,49 @@ You may also run inference from any of the [saved training checkpoints](#inferen
...
+
+`--text_encoder_use_attention_mask`: T5 passes the attention mask to the text encoder.
+
+### Tips and Tricks
+We find LoRA to be sufficient for finetuning the stage I model as the low resolution of the model makes representing finegrained detail hard regardless.
+
+For common and/or not-visually complex object concepts, you can get away with not-finetuning the upscaler. Just be sure to adjust the prompt passed to the
+upscaler to remove the new token from the instance prompt. I.e. if your stage I prompt is "a sks dog", use "a dog" for your stage II prompt.
+
+For finegrained detail like faces that aren't present in the original training set, we find that full finetuning of the stage II upscaler is better than
```
Comment (sayakpaul) [inline docs/source/en/training/dreambooth.mdx:540]: Nice info!

I would enumerate the tips and tricks to improve readability. 

Also, if we can include how the upscaler is supposed to be fine-tuning or even just leave a link that shows it, that would be info-complete.

**Review by sayakpaul — APPROVED** [review]
Changes LGTM! I left some minor nits. 

If possible, I would spend one/two sentences to let the users know what does stage II entail. If it's exactly same as stage I, then nope, otherwise, I think including it makes sense.

File: docs/source/en/training/dreambooth.mdx (hunk 1 above)
Comment (patrickvonplaten) [inline docs/source/en/training/dreambooth.mdx:520]:
```suggestion
pipe.scheduler.register_to_config(variance_type="fixed_small")
```
should suffice no?

File: docs/source/en/training/dreambooth.mdx [hunk 4]
```diff
@@ -527,16 +567,57 @@ accelerate launch train_dreambooth_lora.py \
...
+
+`--learning_rate=1e-6`: Lower learning rate than stage I.
+
+`--resolution=256`: The upscaler expects higher resolution inputs
+
+```sh
+export MODEL_NAME="DeepFloyd/IF-II-L-v1.0"
+export INSTANCE_DIR="dog"
+export OUTPUT_DIR="dreambooth_dog_upscale"
+export VALIDATION_IMAGES="image_1.png image_2.png image_3.png image_4.png"
```
Comment (patrickvonplaten) [inline docs/source/en/training/dreambooth.mdx:589]:
```suggestion
export INSTANCE_DIR="dog"
export OUTPUT_DIR="dreambooth_dog_upscale"
export VALIDATION_IMAGES="image_1.png image_2.png image_3.png image_4.png"
```

what is image_1.png ... ? How can we download those . It would be nice if someone that reads the ## IF section can just execute commands to get it running

File: docs/source/en/training/dreambooth.mdx [hunk 5]
```diff
@@ -549,17 +630,50 @@ accelerate launch train_dreambooth.py \
...
+
+`--learning_rate=1e-8`: Even lower learning rate.
+
+`--resolution=256`: The upscaler expects higher resolution inputs
+
+```sh
+export MODEL_NAME="DeepFloyd/IF-II-L-v1.0"
+export INSTANCE_DIR="dog"
+export OUTPUT_DIR="dreambooth_dog_upscale"
+export VALIDATION_IMAGES="image_1.png image_2.png image_3.png image_4.png"
```
Comment (patrickvonplaten) [inline docs/source/en/training/dreambooth.mdx:658]: Same here where do the dirs and images come from? Let's make it easy for the user

File: docs/source/en/training/dreambooth.mdx [hunk 6]
```diff
@@ -549,17 +630,50 @@ accelerate launch train_dreambooth.py \
...
+  --learning_rate=1e-8 \
+  --max_train_steps=2000 \
+  --validation_prompt="a sks dog" \
+  --validation_steps=150 \
+  --checkpointing_steps=500 \
+  --pre_compute_text_embeddings \
+  --tokenizer_max_length=77 \
+  --text_encoder_use_attention_mask \
+  --validation_images $VALIDATION_IMAGES \
+  --class_labels_conditioning timesteps
```
Comment (patrickvonplaten) [inline docs/source/en/training/dreambooth.mdx:678]: The default should always be to push to the Hub. Let's add this here and make sure it works:
```suggestion
  --class_labels_conditioning timesteps \
  --push_to_hub
```

File: docs/source/en/training/dreambooth.mdx [hunk 7]
```diff
@@ -549,17 +630,50 @@ accelerate launch train_dreambooth.py \
...
-  --pre_compute_text_embeddings \ # Pre compute text embeddings to that T5 doesn't have to be kept in memory
+  --text_encoder_use_attention_mask \
+  --tokenizer_max_length 77 \
+  --pre_compute_text_embeddings \
   --use_8bit_adam \ # 
   --set_grads_to_none \
-  --skip_save_text_encoder # do not save the full T5 text encoder with the model
-```
\ No newline at end of file
+  --skip_save_text_encoder
```
Comment (patrickvonplaten) [inline docs/source/en/training/dreambooth.mdx:645]:
```suggestion
  --skip_save_text_encoder \
  --push_to_hub
```

By default we should always push to the Hub

File: docs/source/en/training/dreambooth.mdx (hunk 1 above)
Comment (williamberman) [reply docs/source/en/training/dreambooth.mdx:520]: I wouldn't recommend mutating existing constructed schedulers since depending on the argument they sometimes change other instance variables. Since they're cheap to construct, I think it's good to always demonstrate freshly constructing a new instance

File: docs/source/en/training/dreambooth.mdx (hunk 3 above)
Comment (williamberman) [reply docs/source/en/training/dreambooth.mdx:540]: Hmm, what should we link to? The stage II fine tuning section is just a bit further on in the page

File: docs/source/en/training/dreambooth.mdx (hunk 4 above)
Comment (williamberman) [reply docs/source/en/training/dreambooth.mdx]: https://github.com/huggingface/diffusers/pull/3628/commits/826d0c38c05a0720faee5ad6f5b2d34e708399c0 done here!
